src/NeoSharp.Core/Persistence/Contexts/BlockHeaderContext.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System.Threading.Tasks;
+using NeoSharp.BinarySerialization;
+using NeoSharp.Core.Models;
+using NeoSharp.Core.Types;
+
+namespace NeoSharp.Core.Persistence.Contexts
+{
+    public class BlockHeaderContext : IBlockHeaderContext
+    {
+        #region Private Fields 
+        private readonly IDbContext<BlockHeader> _dbContext;
+        private readonly IBinarySerializer _serializer;
+        private readonly IBinaryDeserializer _deserializer;
+
+        #endregion
+
+        #region Constructor 
+        public BlockHeaderContext(
+            IDbContext<BlockHeader> dbContext,
+            IBinarySerializer serializer,
+            IBinaryDeserializer deserializer)
+        {
+            _dbContext = dbContext;
+            _serializer = serializer;
+            _deserializer = deserializer;
+        }
+        #endregion
+
+        #region IBlockHeaderContext implementation
+        public Task Add(BlockHeader blockHeader)
+        {
+            var key = blockHeader.Hash.BuildKey(DataEntryPrefix.DataBlock);
+            var content = _serializer.Serialize(blockHeader);
```

Comment:
Maybe is better send the object without serialize it. If you want to store it on Mongo (for example) you need a Json

Reply:
Then the decision of serialize or not should be from the specific component and not from our wrapper ...

make sense.

Reply:
The decision should be up to storage (dbContext) and the serializer should be inside of it.

Reply:
done .. the decision is made by the model class in the new concept